src/gui/transferlistwidget.cpp

```diff
@@ -428,65 +428,56 @@ void TransferListWidget::previewSelectedTorrents()
 
 void TransferListWidget::setDlLimitSelectedTorrents()
 {
-    QList<BitTorrent::TorrentHandle *> selected_torrents;
-    bool first = true;
-    bool all_same_limit = true;
+    QList<BitTorrent::TorrentHandle *> TorrentsList;
     foreach (BitTorrent::TorrentHandle *const torrent, getSelectedTorrents()) {
-        if (!torrent->isSeed()) {
-            selected_torrents << torrent;
-            // Determine current limit for selected torrents
-            if (first)
-                first = false;
-            else if (all_same_limit && (torrent->downloadLimit() != selected_torrents.first()->downloadLimit()))
-                all_same_limit = false;
-        }
+        if (torrent->isSeed())
+            continue;
+        TorrentsList += torrent;
     }
+    if (TorrentsList.empty()) return;
 
-    if (selected_torrents.empty()) return;
+    int oldLimit = TorrentsList.first()->downloadLimit();
+    foreach (BitTorrent::TorrentHandle *const torrent, TorrentsList) {
```

Comment:
Potentially re-iterating the whole list isn't optimal. Worst-case scenario user selects all his torrent (1000) that have the same limit.
I think this check should stay in the 1st loop.